solutions/level1/cart.rb

```diff
@@ -3,23 +3,26 @@
 require 'json'
 
 class Cart
-  attr_accessor :total, :item_count
+  attr_accessor :total, :item_count, :item_list
 
   def initialize
     @total = 0
     @item_count = 0
+    @item_list = []
   end
 
-  def add_item(book)
-    @total += book.price
-    @item_count += 1
+  def add_item(item)
+    @item_list << item if item != nil
   end
 
-  def generate_receipt(books)
-    parsed_books = []
-    books.each do |book|
-      parsed_books << book.to_json
+  def calculate_total_and_count
```

Comment:
do you think `total` and `item_count` could be 2 methods on the Cart object ? would it make sense ?
This method does multiple things and that not in the spirit of single responsibility